Lib/functools.py

```diff
@@ -302,9 +299,8 @@ def __call__(*args, **keywords):
         if not args:
             raise TypeError("descriptor '__call__' of partial needs an argument")
         self, *args = args
-        newkeywords = self.keywords.copy()
-        newkeywords.update(keywords)
-        return self.func(*self.args, *args, **newkeywords)
+        keywords = {**self.keywords, **keywords}  # allow duplicates
```

Comment:
Slightly confused over why allow duplicate comment is added here since dict merge removes duplicates though `self.keywords` and `keyword` might have same keys.

Reply:
I wanted to mention the reason the reason of why we cannot just call `self.func(*self.args, *args, **self.keywords, **keywords)`. The latter syntax does not allow an argument name be duplicated in `self.keywords` and `keywords`.

I'll clarify the comments.